### PR TITLE
Store SHA/version of synced repos in VMR using `.props` files

### PR DESCRIPTION
**Pull request by premun**
Tag files are no longer used and commit SHA / version for each synchronized individual repository is stored in/read from:
- `git-info/AllRepoVersions.props` file
- `git-info/[repo].props`

`VmrManager` now also reads the package versions from `Version.Details.xml` and stores it too.

These files are the same as Source-Build tarball uses ([link](https://github.com/dotnet/installer/tree/main/src/SourceBuild/Arcade/src)). The logic is the same, only uses `Version.Details.xml` reader from `DarcLib`. The tarball tasks could utilize this and code could be removed from `dotnet/installer` if we want.

Resolves https://github.com/dotnet/arcade/issues/10264

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/src/DarcLib/Models/MsBuildPropsFile.cs [hunk 1]
```diff
@@ -48,6 +49,12 @@ protected MsBuildPropsFile(bool? orderPropertiesAscending)
 
     public void SerializeToXml(string path)
     {
+        var parentDir = Path.GetDirectoryName(path) ?? throw new ArgumentException($"'{path}' is not a valid path.");
```
Comment (tkapin) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Models/MsBuildPropsFile.cs:52]: Nit: the `parentDir` is a bit misleading name as it's actually the name of the directory, not it's parent.

File: src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs [hunk 2]
```diff
@@ -36,21 +36,43 @@ public AllVersionsPropsFile(Dictionary<string, string> versions)
...
     {
         var key = SanitizePropertyName(repository) + ShaPropertyName;
         Versions.TryGetValue(key, out var sha);
 
         key = SanitizePropertyName(repository) + PackageVersionPropertyName;
         Versions.TryGetValue(key, out var version);
+
+        if (sha is null && version is null)
+        {
+            return null;
```
Comment (tkapin) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs:49]: Out of curiosity, are there any guidelines to return null when using tuples in C#? I.e., null ofr each of the return values vs. using the null for the whole tuple.

File: src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs [hunk 3]
```diff
@@ -36,21 +36,43 @@ public AllVersionsPropsFile(Dictionary<string, string> versions)
...
+        }
+
         return (sha, version);
     }
 
-    public void UpdateVersion(string repository, string sha, string version)
+    public void UpdateVersion(string repository, string? sha, string? version)
```
Comment (tkapin) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs:55]: Should be covered by tests

**Review by tkapin — DISMISSED** [review]

File: src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs (hunk 3 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs]: The class has tests already. Or do you mean cover the null values?

File: src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs (hunk 2 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs:47]: I have created an enclosing type for the tuple

File: src/Microsoft.DotNet.Darc/src/DarcLib/Models/MsBuildPropsFile.cs (hunk 1 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/src/DarcLib/Models/MsBuildPropsFile.cs:52]: Renamed

File: src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs (hunk 3 above)
Comment (tkapin) [reply src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs]: That's right, but you are adding a null-branching logic which doesn't seem to be reflected in the tests.

**Review by tkapin — DISMISSED** [review]

File: src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs (hunk 3 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs]: Not anymore (with the enclosing type name)

Comment (premun) [reply src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs]: Actually yes, for the respective items inside. I will add a test

Comment (premun) [reply src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs]: I have disallowed null as it's not a valid use case anyway.

**Review by tkapin — APPROVED** [review]

**Pull request merged**